service.c

```diff
@@ -73,8 +73,9 @@ VOID ServiceStart(DWORD dwArgc, LPTSTR *lpszArgv)
 		goto cleanup;
 	}
 
-	while(1)
+	while(ssStatus.dwCurrentState == SERVICE_RUNNING)

```

Comment:
Does it actually work? I didn't check, but I think that it's better to wait for `hServerStopEvent`. That's also what is done in the [Microsoft example](https://github.com/microsoft/Windows-classic-samples/blob/1d363ff4bd17d8e20415b92e2ee989d615cc0d91/Samples/Win7Samples/winbase/service/Simple.c).